### PR TITLE
[1652] Add apply trainee data page

### PR DESCRIPTION
**Pull request by FriedSock**
### Context

https://trello.com/c/8aqWACnk/1652-m-apply-review-application-page

This page is similar to the check details page in that there will be
Subsections of data that come from an apply application. The difference
being that in a future iteration, there will be error validation that
will point the user to fields they need to correct, rather than just
being told a section is incomplete and having to navigate through
every step of the form.

Given the similarities with the check details page (and trn submission
Form), I tried to share some behaviour between them, but decided to go
for a bit of copying until we add more functionality, namely:
- Error checking on the apply trainee data page
- Apply draft version of the check details page

### Changes proposed in this pull request

- Adds a new page for apply trainee data (to be linked from the review-draft page for apply draft trainees, containing sections for personal details, contact details, diversity, degrees
- Sets this page as an origin in the page tracker, so change links into the subsections will return to this page
- If the checkbox "I have reviewed this" is checked, all 4 sections are completed.

### Guidance to review

- Create a draft trainee with some data for the above 4 sections
- visit `/trainees/{trainee_id}/apply-trainee-data/edit`
- try changing attributes and check that you are returned to the page
- try submitting the form with/without checking the "I have reviewed" checkbox

Review app link https://rtt-review-pr-875.herokuapp.com/

### CODE REVIEW

**Comment by FriedSock** [comment]
[[M] Apply review application page](https://trello.com/c/8aqWACnk/1652-m-apply-review-application-page)

**Comment by FriedSock** [comment]
> https://rtt-review-pr-864.herokuapp.com/trainees/DgUN2p75ksioX2AEgEzDPuC2/apply-trainee-data/edit
> 
> image
> 
> Should it be showing this

fixed

File: app/controllers/trainees/apply_trainee_data_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    def update
+      if params[:apply_trainee_data_form][:mark_as_reviewed] == "1"
+        @form = ApplyTraineeDataForm.new(trainee: @trainee)
+        unless @form.save
+          render :edit
+        end
+      end
+      redirect_to trainee_path(trainee)
```
[the remark this answers is not in the document]
Comment (FriedSock) [reply app/controllers/trainees/apply_trainee_data_controller.rb:19]: It works in the same way as confirming sections, we confirm things based on that checkbox being submitted, othwerwise just go back to the review-draft page.

File: app/components/trainees/apply_trainee_data/view.html.erb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    <%= render Trainees::Sections::View.new(trainee: @trainee, form: @form, section: :contact_details, show_incomplete: true) %>
+
+    <%= render Trainees::Sections::View.new(trainee: @trainee, form: @form, section: :diversity, show_incomplete: true) %>
+
+    <h2 class="govuk-heading-m">
+      Education
+    </h2>
+
+    <%= render Trainees::Sections::View.new(trainee: @trainee, form: @form, section: :degrees, show_incomplete: true) %>
```
Comment (defong) [inline app/components/trainees/apply_trainee_data/view.html.erb:47]: As `delete` is an option 
image

It now looks weird 
image

**Review by defong — CHANGES_REQUESTED** [review]
The `form`, ie `ApplyTraineeDataForm`, `TrnSubmissionForm`, should do all the hard work of making `Trainees::Sections::View` show the correct underlaying confirmation view or incomplete view for the respective section.

See https://en.wikipedia.org/wiki/Mediator_pattern

**Comment by FriedSock** [comment]
Closing to try and retrigger a migration on the review app

**Review by defong — APPROVED** [review]

**Comment by GeoffJohnson** [comment]
All good

**Pull request merged**